test/e2e/llmisvc/test_llm_inference_service.py

```diff
@@ -0,0 +1,289 @@
+# Copyright 2025 The KServe Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+import time
+
+import pytest
+import requests
+from kubernetes import client
+from kserve import KServeClient, V1alpha1LLMInferenceService, constants
+
+from .test_configs import (
+    LLMINFERENCESERVICE_CONFIGS,
+    generate_test_id,
+    llm_config_factory,
+    KSERVE_TEST_NAMESPACE,
+)
+
+KSERVE_PLURAL_LLMINFERENCESERVICE = "llminferenceservices"
+
+
+@pytest.mark.llminferenceservice(type="cpu")
```

Comment:
I am wondering if we could make this not a marker, but part of parameterized test case and filter out based on external config. i.e. skip this test case if `TEST_MODE` does not have e.g. `cpu` in it.